verilog/tools/kythe/kythe_facts_extractor.cc

```diff
@@ -261,19 +268,51 @@ void KytheFactsExtractor::Visit(const IndexingFactNode& node,
 
 void KytheFactsExtractor::ConstructFlattenedScope(const IndexingFactNode& node,
                                                   const VName& vname,
-                                                  const Scope& current_scope) {
+                                                  Scope& current_scope) {
   const auto tag = node.Value().GetIndexingFactType();
 
   // Determines whether to add the current scope to the scope context or not.
   switch (tag) {
     case IndexingFactType::kFile:
     case IndexingFactType::kModule:
+    case IndexingFactType::kStructOrUnion:
     case IndexingFactType::kClass:
     case IndexingFactType::kMacro:
     case IndexingFactType::kPackage: {
       scope_resolver_->MapSignatureToScope(vname.signature, current_scope);
       break;
     }
+    case IndexingFactType::kVariableDefinition: {
+      // Break if this variable has no type.
+      if (node.Parent() == nullptr ||
+          node.Parent()->Value().GetIndexingFactType() !=
+              IndexingFactType::kDataTypeReference) {
+        scope_resolver_->MapSignatureToScope(vname.signature, current_scope);
+        break;
+      }
+
+      // TODO(minatoma): refactor this and the below case into function.
+      // TODO(minatoma): consider getting rid of kModuleInstance and
+      // kClassInstance and use kVaraibleDefinition if they don't provide
+      // anything new.
+      const std::vector<const VName*> found_vnames =
+          scope_resolver_->SearchForDefinitions(
+              {node.Parent()->Value().Anchors()[0].Value()});
+
+      if (found_vnames.empty()) {
+        break;
+      }
+
+      const Scope* type_scope =
+          scope_resolver_->SearchForScope(found_vnames[0]->signature);
+      if (type_scope != nullptr) {
+        current_scope.AppendScope(*type_scope);
```

Comment:
This section needs explanation: why is this this injecting or modifying `current_scope`?  is this the right place to be doing that?

Reply:
For structs like this:
```
struct {
int x;
} var1;
```

`var1` is a variable definitions and at the same time contains other variable definitions inside it.
So we have to modify it's scope to get all the variables it should have as members.

but anyway scope resolver will be refactored I am just waiting after the PRs.
but this will be addressed in a follow up PR.

Reply:
Is it possible to create an anonymous (auto-generated/auto-enumerated internal name) struct scope and finish it before moving onto `var1`?

Reply:
yes this was the other way around. 
will fix in a coming PR.